### ISSUE TITLE
Error when downloading subtitles for subfolders.

### ISSUE THREAD

**Issue by thomasloldk** [Diaoul/subliminal#775]
Hey guys ! , I'm no programmer or python expert but I usually know my way arround. these kind of stuff

When im running the script it collets videos , But is unable to download the subs.
`~$ subliminal download -l da tv/the\ big\ bang\ theory/The\ Big\ Bang\ Theory\ Season\ 09\ \(720p\)\ S09\ Complete\ 9\ MP4\ AAC\ eSub\ HDTV\ \[KoTuWa\]/* Collecting videos [####################################] 100% 24 videos collected / 0 video ignored / 3 errors Downloading subtitles [#-----------------------------------] 4% The.Big.Bang.Theory.S09E01.720p.HDTV.[KoTuWa]x.mp4 Traceback (most recent call last): File "/home/thomas/.local/bin/subliminal", line 11, in <module> sys.exit(subliminal()) File "/home/thomas/.local/lib/python3.5/site-packages/click/core.py", line 722, in __call__ return self.main(*args, **kwargs) File "/home/thomas/.local/lib/python3.5/site-packages/click/core.py", line 697, in main rv = self.invoke(ctx) File "/home/thomas/.local/lib/python3.5/site-packages/click/core.py", line 1066, in invoke return _process_result(sub_ctx.command.invoke(sub_ctx)) File "/home/thomas/.local/lib/python3.5/site-packages/click/core.py", line 895, in invoke return ctx.invoke(self.callback, **ctx.params) File "/home/thomas/.local/lib/python3.5/site-packages/click/core.py", line 535, in invoke return callback(*args, **kwargs) File "/home/thomas/.local/lib/python3.5/site-packages/click/decorators.py", line 27, in new_func return f(get_current_context().obj, *args, **kwargs) File "/home/thomas/.local/lib/python3.5/site-packages/subliminal/cli.py", line 403, in download hearing_impaired=hearing_impaired, only_one=single) File "/home/thomas/.local/lib/python3.5/site-packages/subliminal/core.py", line 211, in download_best_subtitles for s in subtitles], key=operator.itemgetter(1), reverse=True) File "/home/thomas/.local/lib/python3.5/site-packages/subliminal/core.py", line 211, in <listcomp> for s in subtitles], key=operator.itemgetter(1), reverse=True) File "/home/thomas/.local/lib/python3.5/site-packages/subliminal/score.py", line 106, in compute_score matches = subtitle.get_matches(video) File "/home/thomas/.local/lib/python3.5/site-packages/subliminal/providers/podnapisi.py", line 52, in get_matches if video.series and sanitize(self.title) == sanitize(video.series): File "/home/thomas/.local/lib/python3.5/site-packages/subliminal/utils.py", line 111, in sanitize string = re.sub(r'[%s]' % re.escape(''.join(characters)), ' ', string) File "/usr/lib/python3.5/re.py", line 182, in sub return _compile(pattern, flags).sub(repl, string, count) TypeError: expected string or bytes-like object`

I have tired searched similar cases and tried their fixex but without luck
I have both tried installaing it systemwide and also just for the user.
Tried both pip3 and pip (2.7) option. . And ofcoruse i installed / uninstalled In ordre.

And i puts out same error everytime.

I'm running . ubuntu server 16.04

thanks in advance (and sorry for my broken English)

**Comment by thomasloldk** [comment]
Same issue on fresh installed Debian 8.8 .. please some help...

`thomas@FA5:~$ sudo subliminal download -l en tv/The\ Big\ Bang\ Theory\ Season\ 09\ \(720p\)\ S09\ Complete\ 9\ MP4\ AAC\ eSub\ HDTV\ \[KoTuWa\]/*
Collecting videos  [####################################]  100%
24 videos collected / 0 video ignored / 3 errors
Downloading subtitles  [#-----------------------------------]    4%  The.Big.Bang.Theory.S09E01.720p.HDTV.[KoTuWa]x.mp4
Traceback (most recent call last):
  File "/usr/local/bin/subliminal", line 9, in <module>
    load_entry_point('subliminal==2.0.5', 'console_scripts', 'subliminal')()
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/subliminal/cli.py", line 403, in download
    hearing_impaired=hearing_impaired, only_one=single)
  File "/usr/local/lib/python2.7/dist-packages/subliminal/core.py", line 211, in download_best_subtitles
    for s in subtitles], key=operator.itemgetter(1), reverse=True)
  File "/usr/local/lib/python2.7/dist-packages/subliminal/score.py", line 106, in compute_score
    matches = subtitle.get_matches(video)
  File "/usr/local/lib/python2.7/dist-packages/subliminal/providers/podnapisi.py", line 52, in get_matches
    if video.series and sanitize(self.title) == sanitize(video.series):
  File "/usr/local/lib/python2.7/dist-packages/subliminal/utils.py", line 111, in sanitize
    string = re.sub(r'[%s]' % re.escape(''.join(characters)), ' ', string)
  File "/usr/lib/python2.7/re.py", line 155, in sub
    return _compile(pattern, flags).sub(repl, string, count)
TypeError: expected string or buffer`

**Comment by fernandog** [comment]
@thomasloldk why you add the slashes to the filename?

Seems like the string is None, so or self.title is None or video.series is None

```
string = None
string = re.sub(r'[%s]' % re.escape(''.join(characters)), ' ', string)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "C:\Python27\lib\re.py", line 155, in sub
    return _compile(pattern, flags).sub(repl, string, count)
TypeError: expected string or buffer
```

**Comment by ratoaq2** [comment]
That's the way bash in Linux escapes a space. Perfectly normal

**Comment by fernandog** [comment]
Why not add double/single quotes around the filename?

**Comment by ratoaq2** [comment]
Because you don't type the path, just type initials + tab and it auto completes in that way.

**Comment by ratoaq2** [comment]
The issue is that there's a subtitle which its name is parsed by guessit and returns multiple titles. 
I bet it's an old guessit version. 
@thomasloldk could you tell us which guessit and rebulk versions you have?

**Comment by thomasloldk** [comment]
@ratoaq2  
Hey Thanks for getting back you me ! 

I seems like i the meantime from when i posted this i got a step closer to fix this. 
Don't know why i could not get the subtitle for the big bang theory. 
But i tested it again , i the issuse only occur when trying to download multiple subtitles at once. 
I downloaded subs. for suicide squad successfully for the first time now. 
I am useing Guessit 2.1.4 - Rebulk 0.9.0 , Downloaded from the default debian repo. 

This is the output: 
`thomas@FA5:~/movies$ subliminal download -l da *
Collecting videos  [####################################]  100%
93 videos collected / 1 video ignored / 0 error
Downloading subtitles  [##########--------------------------]   29%  0d 00:00:47  Inferno.2016.1080p.BluRay.x264-[YTS.AG].mp4
Traceback (most recent call last):
  File "/usr/local/bin/subliminal", line 9, in <module>
    load_entry_point('subliminal==2.0.5', 'console_scripts', 'subliminal')()
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/subliminal/cli.py", line 403, in download
    hearing_impaired=hearing_impaired, only_one=single)
  File "/usr/local/lib/python2.7/dist-packages/subliminal/core.py", line 211, in download_best_subtitles
    for s in subtitles], key=operator.itemgetter(1), reverse=True)
  File "/usr/local/lib/python2.7/dist-packages/subliminal/score.py", line 106, in compute_score
    matches = subtitle.get_matches(video)
  File "/usr/local/lib/python2.7/dist-packages/subliminal/providers/opensubtitles.py", line 98, in get_matches
    matches |= guess_matches(video, guessit(self.movie_release_name, {'type': 'movie'}))
  File "/usr/local/lib/python2.7/dist-packages/subliminal/subtitle.py", line 235, in guess_matches
    if video.format and 'format' in guess and guess['format'].lower() == video.format.lower():
AttributeError: 'list' object has no attribute 'lower'`

**Comment by ratoaq2** [comment]
This traceback is different from the previous one.

Now the issue is exactly this one: https://github.com/Diaoul/subliminal/issues/761

The problem is that subliminal doesn't expect that the subtitle name parser will return multiple values for a single field, in this case: `format`

**Comment by thomasloldk** [comment]
@ratoaq2 

I tried to install the quick fix from the comments #769 , without luck. 
So I'm able to download multiple subtitles anymore ? 
Please explain a little furtuer if you mind.

Thanks so far :-)

**Comment by thomasloldk** [comment]
@ratoaq2 
After some more debugging i found out that the error occurs  because of sub folders with the movies located in it. 
Like: movies/movie/film.mp4

But i work like a charm when: movies/film.mp4

**Comment by thomasloldk** [comment]
Subliminal still error when multple augments is in the filename. 
Like movie.blueray.1080p.mp4
But works with , movie.1080p.mp4

**Comment by Madseason41** [comment]
I have the same issue, subtitle download works when calling the file directly, but not when calling on subliminal to run on a directory. I get no errors, subliminal just doesn't download any subtitles.

**Comment by Diaoul** [comment]
This is probably due to guessit way to parse the video file path. I would suggest to run subliminal from within the "movies" directory so it parses the correct part i.e. "movienamefolder/file.mp4".

**Comment by CyberMew** [comment]
Does guessit work differently with different providers?

**Comment by Madseason41** [comment]
I ended up running guessit directly on the file, and it turns out my folder naming scheme was causing issues. Also I was messing around with sickbeard_mp4_automator and I had my guessit version out of whack.

I ended up downgrading guessit, and changing my folder scheme, now all is well.

**Comment by ratoaq2** [comment]
Could you please open an issue in guessit with the problematic input and guessit version?

**Comment by nomadturk** [comment]
I also had a similar problem after messing around with sickbeard_mp4_automator.

I had to run these to commands to get subliminal going again:

```
pip uninstall "guessit<2"
pip install "guessit>2"
```

**Comment by ratoaq2** [comment]
Fixed by #761
